src/caffe/util/im2col.cu

```diff
@@ -136,67 +160,69 @@ template <typename Dtype>
 void im2col_nd_gpu(const Dtype* data_im, const int num_spatial_axes,
     const int num_kernels, const int* im_shape, const int* col_shape,
     const int* kernel_shape, const int* pad, const int* stride,
-    Dtype* data_col) {
+    const int* dilation, Dtype* data_col) {
+  // num_axes should be smaller than block size
+  DCHECK_LT(10, CAFFE_CUDA_NUM_THREADS);
```

Comment:
Should this be `DCHECK_LE`? (I also think we should probably check `num_spatial_axes` here instead of `10`, to lose the constant literal which is subject to change.)